src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +11,26 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        Lock writeLock = lock.writeLock();
+        writeLock.lock();
+        try {
+            list.add(element);
+        } finally {
+            writeLock.unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        Lock readLock = lock.readLock();
+        readLock.lock();
+        try {
+            return list.get(index);
+        } finally {
+            readLock.unlock();
+        }
     }
 
     public int size() {
-        // write your code here
-        return 0;
+        return list.size();
```

Comment:
The `size` method is not protected by a read lock. In a concurrent environment, this could lead to inconsistent results if the list is being modified by another thread. Consider using a read lock to ensure thread safety.